python/mxnet/ndarray/ndarray.py

```diff
@@ -1854,8 +1854,10 @@ def full(shape, val, ctx=None, dtype=mx_real_t, out=None):
     >>> mx.nd.full((1, 2), 2.0, dtype='float16').asnumpy()
     array([[ 2.,  2.]], dtype=float16)
     """
-    out = empty(shape, ctx, dtype) if out is None else out
-    out[:] = val
+    if ctx is None:
+        ctx = Context.default_ctx
+    dtype = mx_real_t if dtype is None else dtype
+    out = _internal._full(shape=shape, ctx=ctx, dtype=dtype, value=val, out=out)
```

Comment:
how about change setitem to use full instead?

Reply:
What is setitem? Is that python code? I’m not especially familiar with the python parts.

Reply:
I looked at __setiem__ in ndarray.py.  It is not clear how full() would work there. Can you elaborate on what you have in mind?

Reply:
I guess that would be replacing some value assignment functions used in `__setitem__` with `_full`.

In order to avoid code conflict, I can make that change since I am working on `__setitem__` right now for advanced indexing.

Reply:
ok

Reply:
@reminisce remember to change this back to out[:] = val after you've made the change